catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java

```diff
@@ -113,6 +119,14 @@ protected void onCreate(Bundle savedInstanceState) {
 			return;
 		}
 
+		if (getIntent().getBooleanExtra(RESTART_INTENT, false)) {
+			Intent restartIntent = getApplicationContext().getPackageManager().getLaunchIntentForPackage(getApplicationContext().getPackageName());
+			PendingIntent intent = PendingIntent.getActivity(getApplicationContext(), 0, restartIntent, PendingIntent.FLAG_CANCEL_CURRENT);
+			AlarmManager manager = (AlarmManager) getApplicationContext().getSystemService(Context.ALARM_SERVICE);
+			manager.set(AlarmManager.RTC, System.currentTimeMillis() + restartDelayAfterAppClose, intent);
+			System.exit(2);
```

Comment:
why 2?
the android documentation states: Termination status. By convention, a nonzero status code indicates abnormal termination.